plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php

```diff
@@ -349,6 +355,31 @@ public function serializeAltNames()
     return $serialized;
   }
 
+  public function serializeObjectRelations()
+  {
+    $relations = array();
+
+    $sql = "SELECT r.type_id, r.object_id, r.subject_id, oo.class_name \r
+              AS object_class, so.class_name AS subject_class \r
+              FROM ".QubitRelation::TABLE_NAME." r \r
+              INNER JOIN ".QubitObject::TABLE_NAME." oo ON r.object_id=oo.id \r
+              INNER JOIN ".QubitObject::TABLE_NAME." so ON r.subject_id=so.id \r
+              WHERE r.object_id=? OR r.subject_id=?";
```

Comment:
Remove the JOINS to the `object` table, add a JOIN to the `term` table on `type_id` and filter by `taxonomy_id` (ACTOR_RELATION_TYPE_ID). This should speed up the SQL query and avoid the filtering in the ES query.

Nitpick, the `\r` are not really needed.